src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -244,6 +244,7 @@ private void RetrieveNavigationPropertyPaths(IEdmNavigationProperty navigationPr
             // append a navigation property.
             currentPath.Push(new ODataNavigationPropertySegment(navigationProperty));
             AppendPath(currentPath.Clone());
+            IEdmEntityType navEntityType = navigationProperty.ToEntityType();
```

Comment:
Nit: We can move this variable declaration and assignment inside the `else` block closer to where it is being used.